src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -1266,6 +1266,12 @@ public void setAutoResizeMode(int mode) {
             autoResizeMode = mode;
             resizeAndRepaint();
             if (tableHeader != null) {
+                if (mode == JTable.AUTO_RESIZE_LAST_COLUMN) {
+                    if (columnModel.getColumnCount() > 0) {
+                        tableHeader.setResizingColumn(
+                                columnModel.getColumn(columnModel.getColumnCount() - 1));
+                    }
```

Comment:
Check added to guard against the possibility of `DefaultTableColumnModel.getColumn` which according to spec can throw AIOBE
` if columnIndex is out of range: (columnIndex < 0 || columnIndex >= getColumnCount())`